controllers/nstemplateset/cluster_resources_test.go

```diff
@@ -500,6 +500,30 @@ func TestPromoteClusterResources(t *testing.T) {
 			})
 		})
 
+		t.Run("upgrade from base to advanced tier by changing only the tier label", func(t *testing.T) {
```

Comment:
well, the purpose of this test is to verify that the CRQ is updated when only the `tier` label is changed and the `templateref` label stays the same. It's to verify that this PR really fixes the problem.
I slightly changed it, so I hope that it will be better now: eae58e1